src/api/v1/db/oracledb/doughs-dao.js

```diff
@@ -265,4 +289,53 @@ const getDoughById = async (id) => {
   }
 };
 
-export { getDoughs, postDough, getDoughById };
+/**
+ * convert object body to an UPDATE query with correct bind params
+ *
+ * @param {object} body a PATCH request body
+ * @returns {object} `bindParams` and `query`
+ */
+const createPatchQueryAndBindParams = (body) => {
+  const inBindparams = getBindParams(body);
+
+  const query = doughsPatchQuery(inBindparams);
+
+  const bindParams = _.assign(
+    inBindparams,
+    doughsOutBindParams,
+    { id: body.data.id },
+  );
+
+  return { query, bindParams };
+};
+
+
+/**
+ * Update a specific dough recipe
+ *
+ * @param {object} body
+ * @returns {Promise<object>} the updated dough
+ */
+const updateDoughById = async (body) => {
+  if (_.isEmpty(body.data.attributes)) return getDoughById(body.data.id);
+
+  const { bindParams, query } = createPatchQueryAndBindParams(body);
+  const connection = await getConnection();
+
+  try {
+    const result = await connection.execute(
+      query,
+      bindParams,
+      { autoCommit: true },
+    );
+
+    const rawDough = convertOutBindsToRawDough(result.outBinds);
+    return serializeDough(rawDough, `doughs/${body.data.id}`);
```

Comment:
If I were you, I'll probably add an additional logic check to make sure the returned ID is equivalent to the ID we pass from the PATCH body.